apps/wallet/src/ui/app/hooks/useGetValidators.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { isSuiObject, isSuiMoveObject } from '@mysten/sui.js';
+import BigNumber from 'bignumber.js';
+import { useMemo } from 'react';
+
+import { useGetObject } from './useGetObject';
+import {
+    STATE_OBJECT,
+    VALDIATOR_NAME,
+} from '_app/staking/usePendingDelegation';
+
+//TODO: Remove when available on SDK, types should come from the SDK
+import type { ValidatorState } from '_app/staking/home/ValidatorDataTypes';
+
+function getStakePercent(stake: bigint, total: bigint): number {
+    const bnStake = new BigNumber(stake.toString());
+    const bnTotal = new BigNumber(total.toString());
+    return bnStake.div(bnTotal).multipliedBy(100).toNumber();
+}
+
+export function useGetValidators(walletAddress: string | null) {
+    const { data, isLoading, isError } = useGetObject(STATE_OBJECT);
+
+    const validatorsData =
+        data && isSuiObject(data.details) && isSuiMoveObject(data.details.data)
+            ? (data.details.data.fields as ValidatorState)
+            : null;
+
+    const totalStake =
+        validatorsData?.validators.fields.total_validator_stake || 0n;
+
+    const validators = useMemo(() => {
+        if (!validatorsData) return [];
+        return validatorsData.validators.fields.active_validators.map((av) => {
+            const rawName = av.fields.metadata.fields.name;
+
+            let name: string;
+
+            if (Array.isArray(rawName)) {
+                name = String.fromCharCode(...rawName);
+            } else {
+                name = Buffer.from(rawName, 'base64').toString();
+                if (!VALDIATOR_NAME.test(name)) {
+                    name = rawName;
+                }
+            }
+            const {
+                sui_balance,
+                starting_epoch,
+                pending_delegations,
+                delegation_token_supply,
+            } = av.fields.delegation_staking_pool.fields;
+
+            const num_epochs_participated =
+                validatorsData.epoch - starting_epoch;
+            const APY =
+                (1 +
+                    (sui_balance - delegation_token_supply.fields.value) /
+                        delegation_token_supply.fields.value) ^
+                (365 / num_epochs_participated - 1);
+
+            const pending_delegationsByAddress = pending_delegations
+                ? pending_delegations.filter(
+                      (d) => d.fields.delegator === walletAddress
+                  )
+                : [];
+
+            return {
```

Comment:
I still remain unconvinced that this is a hook that needs to exist.

Reply:
We use most of this base meta information across multiple components. Do you mean I repeat the APY calculation across components?

Reply:
Yeah, or just have this hook do the plain lookup of validators in the system object. Otherwise it seems better suited to move into the display logic. Otherwise these ends up being effectively view-models, which we don't want. We want to pass around the raw underlying data when we can because it makes it a lot easier to build out.